go/vt/vttablet/tabletmanager/action_agent.go

```diff
@@ -290,7 +290,10 @@ func NewActionAgent(
 	// The db name is set by the Start function called above
 	agent.VREngine = vreplication.NewEngine(ts, tabletAlias.Cell, mysqld, func() binlogplayer.DBClient {
 		return binlogplayer.NewDBClient(agent.DBConfigs.FilteredWithDB())
-	}, agent.DBConfigs.FilteredWithDB().DbName)
+	},
+		agent.DBConfigs.ExternalReplWithDB(),
+		agent.DBConfigs.FilteredWithDB().DbName,
```

Comment:
This looks odd. We need to think of something else.

Reply:
I fixed this.